.github/workflows/run_contract_tests.yaml

```diff
@@ -17,64 +17,220 @@ jobs:
           rustup override set 1.61
 
       - name: Install `wasm32-unknown-unknown`
-        run: rustup target add wasm32-unknown-unknown      
+        run: rustup target add wasm32-unknown-unknown    
 
-      - name: Run light client tests
-        run: cd contracts/light_client; cargo test -- --nocapture
-
-      - name: Run prover tests
+      - name: Build Contracts
         run: |
-          cd contracts/mock_light_client
+          cd contracts/
+          cd mock_light_client
           ./build.sh
-          cd ../prover
+          cd ../mock_prover
           ./build.sh
-          cargo test -- --nocapture
+          cd ../..
 
-      - name: Run types tests
-        run: cd contracts/types; cargo test -- --nocapture
+      - name: Move builds
+        run: |
+          cp contracts/mock_prover/target/wasm32-unknown-unknown/release/mock_prover.wasm .
+          cp contracts/mock_light_client/target/wasm32-unknown-unknown/release/mock_light_client.wasm .
 
-      - name: Run connector_permissions tests
+      - name: Store builds
+        uses: actions/upload-artifact@v3
+        with:
+          name: mocks
+          path: |
+            mock_prover.wasm
+            mock_light_client.wasm
+
+  test_rust:
+    name: Rust test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Sources
+        uses: actions/checkout@v3
+
+      - name: Set rust version
         run: |
-          cd contracts/connector_permissions
-          ./build.sh
-          cargo test -- --nocapture
+          rustc --version
+          rustup install 1.61
+          rustup override set 1.61
+
+      - name: Install `wasm32-unknown-unknown`
+        run: rustup target add wasm32-unknown-unknown    
 
-      - name: Run ft_connector tests
+      - name: Run Light Client Test
         run: |
-          cd contracts/ft_bridge_token
-          ./build.sh
-          ./deploy.sh
-          cd ../bridge_token_deployer
-          ./build_ft.sh
-          cd ../mock_prover
-          ./build.sh
-          cd ../connector_permissions
-          ./build.sh
-          cd ../ft_connector
-          ./build.sh
+          cd contracts/light_client 
+          cargo test -- --nocapture
+          cd ../types
```

Comment:
would it make sense to move test_light_client as a separate job that depends on test_rust what could be renamed to rust_setup?

Reply:
Nothing to gain there, will leave it, but split steps for LC and types tests